linode/helper/frameworkfilter/local_filter.go

```diff
@@ -44,7 +46,7 @@ func (f Config) matchesFilter(
 		filterName := filter.Name.ValueString()
 
 		// Skip if this field should be filtered at an API level
-		if f[filterName].APIFilterable {
+		if f[filterName].APIFilterable && checkFilterSkip(filter) {
```

Comment:
May I know why calling `checkFilterSkip` here is necessary here?

Reply:
This check is necessary as some fields may be API-filterable but are not using an `exact` matching type, so we should only skip local filtering on fields that are both API filterable and have `exact`/nil values for `match_by`.